harvard-huit/Harvard-Geospatial-Library
# Upgrading to passenger-ruby26:2.0.0. Updating tmp path in webapp.conf.

**Upgrading to passenger-ruby26:2.0.0. Updating tmp path in webapp.conf.**
* * *

**JIRA Ticket**: None

* Other Relevant Links (Mailing list discussion, related pull requests, etc.)
- https://letsencrypt.org/docs/dst-root-ca-x3-expiration-september-2021/
- https://github.com/phusion/passenger-docker/issues/306#issuecomment-932822824
- https://stackoverflow.com/questions/66977603/passenger-6-0-8-loaded-as-an-nginx-module-fails-to-start-up

# What does this Pull Request do?
When building HGL locally today, the following error occurs:
`#7 0.787 Err:10 https://oss-binaries.phusionpassenger.com/apt/passenger focal Release
#7 0.787   Certificate verification failed: The certificate is NOT trusted. The certificate chain uses expired certificate.  Could not handshake: Error in the certificate verification. [IP: 109.107.35.58 443]`

This is because the phusionpassenger.com site uses LetsEncrypt for it's secure certificates. Their DST Root CA X3 expired on September 30, and has been replaced with ISRG Root X1 (See LetsEncrypt announcement in Relevant Links above).

Upgrading our base-image for passenger-ruby26 fixes this issue. Here are the latest tags on the passenger-ruby26 image:
https://hub.docker.com/r/phusion/passenger-ruby26/tags?page=1&ordering=last_updated

Note that 2.0.0 was released 3 days ago, which corresponds with the LetsEncrypt deadline. I tried upgrading just to 1.0.19 (the latest 1.0 release) and the build still fails. It appears upgrading to 2.0.0 is the only way to go.

After upgrading the following error was thrown by nginx:
`nginx: [emerg] mkdir() "/usr/share/nginx/passenger_temp" failed (13: Permission denied)`

I solved this with the [help of StackOverflow](https://stackoverflow.com/questions/66977603/passenger-6-0-8-loaded-as-an-nginx-module-fails-to-start-up)  by changing Passenger's temp log location in the `webapp.conf` file. Note that that file is not committed to our repositories and is a configuration file. **You will need to manually update that file locally, as well as on the dev / qa / prod servers as we deploy.** The changed `webapp.conf.template` file in this PR will show you the change you need to make.

I have updated `docker-compose.yml` with a new tag to deploy this up. _I have successfully built the image locally, but have not pushed it to the registry._ 

# How should this be tested?

A description of what steps someone could take to:
* Manually update your `webapp.conf` file with the new line in `webapp.conf.template`
* Rebuild the container completely (docker-compose -f docker-compose-local.yml up -d --build --force-recreate)
* Confirm that HGL loads and works as expected.

# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests? No
- integration tests? No

# Interested parties
@awoods @enriquediaz
